packages/logging/src/entry.js

```diff
@@ -87,6 +114,29 @@ function Entry(metadata, data) {
     timestamp: new Date()
   }, metadata);
 
+  if (this.metadata.insertId === undefined) {
+    // JavaScript date has a very coarse granularity (millisecond), which makes
+    // it quite likely that multiple log entries would have the same timestamp.
+    // The Logging API doesn't guarantee to preserve insertion order for entries
+    // with the same timestamp. The service does use `insertId` as a secondary
+    // ordering for entries with the same timestamp. `insertId` needs to be
+    // globally unique (within the project) however.
+    //
+    // We generate an `insertId` that is going to uniquely identify the log
+    // entry across the network of services. It is composed of a ServiceId (a
+    // UUID in our case) along with an incrementing 'event counter'. The other
+    // alternative would be to use `process.hrtime()` as the event id/timestamp,
+    // but that would be more overhead.
+    insertCounter++;
+
+    // We encode in base36 to make the string compact. We left pad with zeros to
+    // make it sortable. We left pad upto a width of 13 as a max 64-bit integer
+    // would fit in 13 digits in base36.
+    var counter = leftPad(insertCounter.toString(36), 13, '0');
+    var insertId = SERVICE_ID + '-' + counter;
```

Comment:
The overall approach looks good. If performance ends up being a problem, I think things can be sped up substantially (~2 orders of magnitude on a simple throughput benchmark) by avoiding string concatenation and incrementing in place:

```javascript
// Initialize
var buffer = Buffer.allocUnsafe(size);
uuid.v4(null, buffer);

// Increment
for (var i = buffer.length - 1; i >= 0; i--) {
  if (buffer[i] !== 255) {
    buffer[i]++;
    break;
  } else {
    buffer[i] = 0;
  }
}

// Serialize
buffer.toString('base64');
```